src/app/Http/Controllers/UploadController.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+
+$count_message;
+$add_user_message;
+
+class UploadController extends Controller
+{
+    /**
+    * Called when /add_user gets a post request from the forms.
+    */
+    public function upload() {
+
+      global $count_message;
+      global $add_user_message;
+
+      if (request()->has('mycsv')) {
+        $this->uploadFile();
+        return redirect()->route('add_user_route')->with('user_count_message', $count_message);
+      }
+      else if (request()->has('add_single_user')) {
+        $this->uploadUser();
+        return redirect()->route('add_user_route')->with('add_message', $add_user_message);
+      }
+      else {
+        return redirect()->route('add_user_route');
+      }
+    }
+
+    /**
+    * Extract data from .csv file.
+    */
+    public function uploadFile() {
```

Comment:
We should also check for parsing errors here too, including invalid emails structure

Reply:
![image](https://user-images.githubusercontent.com/43078156/107840773-fe581800-6d8b-11eb-88d6-09d60457cc6a.png)
Added check for invalid email structure.